### PR TITLE
Add Dockerfile, which installs applications + dependencies.

### PR DESCRIPTION
**Pull request by keeb-zz**
This Dockerfile installs the current working directory on an `Ubuntu 12.10` base image and gets it ready for execution.

### CODE REVIEW

**Comment by keeb-zz** [comment]
Example run is `docker run -e TINGE_FLASK_HOST=0.0.0.0 -e BLEND_FLASK_HOST=0.0.0.0 -e PYTHONPATH=/website keeb/margarine /website/start.sh`

Docker will expose ports 5050 and 5000 to the Docker Host, and allow full routing to the APPs via NAT.

Example is running at http://stinemat.es:49422

**Comment by alunduil** [comment]
I might need a bit more of a rundown on how docker works but shouldn't we have one for blend and one for tinge?  

Also, is it possible to read a configuration file from inside a docker container?  If so, we should probably pass the parameters in via the configuration files.

**Comment by keeb-zz** [comment]
>  but shouldn't we have one for blend and one for tinge?

Ideally, yes. You'd probably want a container for each process. But as the code exists now, running everything inside of 1 container makes most sense as `tinge` and `blend` share the same code base and do not appear to scale independently. I could be wrong on that.

As for configuration, there's a couple ways to handle that...
- Pass in environment variables during `docker run` 

This is preferred as it allows the operator of the image to have control over the "ops"
- Hard code environment variables during build time

Not preferred, but can be overwritten via `docker run`
- Pass in configuration files at buildtime.

Using the `add` step, you can basically place a file at any location for configuration.
- Pass in configuration via volume mounts.

Basically make the host directory available inside of the Docker container. Any changes to the host directory are made available in the container. The downside of this is portability between hosts (as you'll have to carry over the configuration / make the same preparations on the new host)

Great questions, please let me know if you'd prefer a different approach and I'll be happy to incorporate it and clean up the commits.

**Comment by alunduil** [comment]
Let's do individual containers as tinge and blend are designed to scale independently but can also be scaled together (depending on how the operator wants to deploy it).

As for the configuration, I think I'm starting to see how docker puts it all together.  Thanks for the nice little overview.  Given the options and what we're going for (flexibility) let's use the environment variables.  Now I'm glad I have full support for all options via environment variables.

I'm also going to create a docker branch for playing with this before merging to master once the containers for tinge and blend have been split.  I assume there will be a third container for spread?
